modules/data.atmosphere/R/tdm_predict_subdaily_function.R

```diff
@@ -0,0 +1,257 @@
+##' Predict Subdaily Meteorology Function
+##' Met downscaling functions that predict the downscaled meteorology 
+# ----------------------------------- 
+# Description
+# -----------------------------------
+##' @title predict.subdaily.function
+##' @family tdm - Temporally Downscale Meteorology
+##' @author Christy Rollinson, James Simkins
+##' @description This script contains the functions that do the heavy lifting in the final
+##'              script of the tdm workflow titled tdm_predict_subdialy_met.R. It uses a linear
+##'              regression approach by generating the hourly values from the daily means of
+##'              the file the user selects to downscale based on the hourly models and betas
+##'              generated by tdm_generate_subdaily_models.R. 
+# ----------------------------------- 
+# Parameters
+# -----------------------------------
+##' @param dat.mod - data to be predicted at the time step of the training data
+##' @param n.ens - number of hourly ensemble members to generate
+##' @param path.model - path to where the training model & betas is stored
+##' @param lags.init - a data frame of initialization paramters to match the data in dat.mod
+##' @param dat.train - the training data used to fit the model; needed for night/day in 
+##'                    surface_downwelling_shortwave_flux_in_air
+# -----------------------------------
+#----------------------------------------------------------------------
+# Begin Script
+#----------------------------------------------------------------------
+
+# Begin Function
+predict.subdaily.function <- function(dat.mod, n.ens, path.model, lags.list = NULL, 
+    lags.init = NULL, dat.train) {
+    
+    # Set progress bar
+    pb.index <- 1
+    pb <- txtProgressBar(min = 1, max = 8, style = 3)
+    setTxtProgressBar(pb, pb.index)
+    
+    # Figure out if we need to extract the approrpiate
+    if (is.null(lags.init)) {
+        lags.init <- lags.list[[unique(dat.mod$ens.day)]]
```

Comment:
since lags.list defaults to NULL, seems like you need a test for ```!is.null(lags.list)``` before this line, and then possibly an error or warning for the else case